R/FilteredData.R

```diff
@@ -1113,45 +1123,53 @@ FilteredData <- R6::R6Class( # nolint
             selected = NULL
           )
           active_slices_ids <- active_slices_id()
-
-          shiny::isolate({
-            interactive_choice_mock <- lapply(
-              slices_interactive(),
-              function(slice) {
+          duplicated_slice_refs <- duplicated_slice_references()
+          interactive_choice_mock <- lapply(
+            slices_interactive(),
+            function(slice) {
+              # we need to isolate changes in the fields of the slice (teal_slice)
+              shiny::isolate({
                 checkbox_group_element(
                   name = session$ns("available_slices_id"),
                   value = slice$id,
                   label = slice$id,
                   checked = if (slice$id %in% active_slices_ids) "checked",
-                  disabled = slice$locked
+                  disabled = slice$locked ||
+                    get_default_slice_id(slice) %in% duplicated_slice_refs &&
+                      !slice$id %in% active_slices_ids
                 )
-              }
-            )
+              })
+            }
+          )
 
-            non_interactive_choice_mock <- lapply(
-              slices_fixed(),
-              function(slice) {
+          non_interactive_choice_mock <- lapply(
+            slices_fixed(),
+            function(slice) {
+              # we need to isolate changes in the fields of the slice (teal_slice)
+              isolate(
                 checkbox_group_element(
                   name = session$ns("available_slices_id"),
                   value = slice$id,
                   label = slice$id,
                   checked = if (slice$id %in% active_slices_ids) "checked",
-                  disabled = slice$locked
+                  disabled = slice$locked ||
+                    get_default_slice_id(slice) %in% duplicated_slice_refs &&
+                      !slice$id %in% active_slices_ids
                 )
-              }
-            )
+              )
+            }
+          )
 
-            htmltools::tagInsertChildren(
-              checkbox,
-              br(),
-              tags$strong("Fixed filters"),
-              non_interactive_choice_mock,
-              tags$strong("Iteractive filters"),
-              interactive_choice_mock,
-              .cssSelector = "div.shiny-options-group",
-              after = 0
-            )
-          })
+          htmltools::tagInsertChildren(
+            checkbox,
+            br(),
+            tags$strong("Fixed filters"),
+            non_interactive_choice_mock,
+            tags$strong("Iteractive filters"),
+            interactive_choice_mock,
```

Comment:
If the `tag$strong` are moved into the function that generates the mock elements, it would pre-emptively solve  this bug:
![image](https://github.com/insightsengineering/teal.slice/assets/78242757/8c1e5a97-a59c-461f-8b35-851b9ad7e54d)